reverse_engineering/helpers/snowflakeHelper.js

```diff
@@ -28,13 +31,25 @@ const connect = async (logger, { host, username, password, authType, authenticat
 
 	logger.log('info', `Connection name: ${name}\nCloud platform: ${cloudPlatform}\nHost: ${host}\nAuth type: ${authType}\nUsername: ${username}\nWarehouse: ${warehouse}\nRole: ${role}`, 'Connection');
 
+	let authPromise;
 	if (authType === 'okta') {
-		return authByOkta(logger, { account, accessUrl, username, password, authenticator, role, warehouse, timeout });
+		authPromise = authByOkta(logger, { account, accessUrl, username, password, authenticator, role, warehouse, timeout });
 	} if (authType === 'externalbrowser') {
-		return authByExternalBrowser(logger, { account, accessUrl, token, proofKey, username, password, role, warehouse, timeout })
+		authPromise = authByExternalBrowser(logger, { account, accessUrl, token, proofKey, username, password, role, warehouse, timeout })
 	} else {
-		return authByCredentials({ account, username, password, role, warehouse, timeout });
+		authPromise = authByCredentials({ account, username, password, role, warehouse, timeout });
 	}
+
+	return await authPromise.catch(err => {
```

Comment:
so, if there is the "can't reach snowflake" issue, I check and try to connect with cloud provider in host name